python/dgl/graphbolt/impl/gpu_cached_feature_store.py

```diff
@@ -0,0 +1,106 @@
+"""Torch-based feature store for GraphBolt."""
+import torch
+
+from dgl.cuda import GPUCache
+
+from ..feature_store import FeatureStore
+
+__all__ = ["GPUCachedFeatureStore"]
+
+
+class GPUCachedFeatureStore(FeatureStore):
+    r"""GPU cached feature store wrapping a fallback feature store."""
+
+    def __init__(self, fallback_store: FeatureStore, cache_size: int):
+        """Initialize GPU cached feature store with a given fallback.
+        Places the GPU cache to torch.cuda.current_device().
```

Comment:
How can user set which device to load to? Will different GPUs share the same cache?

Reply:
`torch.cuda.set_device(device)` before constructing the gpu cached feature store will ininitialize the GPU cache at the currently set device. Do we want the capability to pass the device, and if it is `None`, we fall back to the current_device?

Reply:
I think `dgl.cuda.GPUCache` doesn't have an option to take in a device argument when constructing the GPU cache. It simply constructs it at the currently set CUDA device in the process, i.e. `CUDA_CALL(cudaGetDevice(&cuda_device));`

Reply:
A single GPUCache instance belongs to a single GPU only. GPUs can't share the same cache. A wrapper for a multi-GPU cache could be implemented though.

Reply:
I see, we can focus on single GPU cases for this PR, for multiple GPU use case, let's re-visit. Please put a comment on the header of the file to clarify it is designed for Single GPU usage.

Reply:
Currently each GPU have a dataloader process, if torch.cuda.current_device() can recognize the device assigned to each process, then it should work properly.

Reply:
https://github.com/dmlc/dgl/blob/b6f5ba9a809fcd2e5b2c440f538c1ccc965a9e59/examples/multigpu/node_classification_sage.py#L242 sets the default device so it will just work out of the box.